spec/data-model/README.md

```diff
@@ -74,28 +120,35 @@ interface CatchallKey {
 ## Patterns
 
 Each `Pattern` represents a linear sequence, without selectors.
-Each element of the sequence MUST have either a `Text` or an `Expression` shape.
-`Text` represents literal _text_,
+Each element of the `body` array MUST either be a string or an `Expression` object.
```

Comment:
```suggestion
Each element of the `body` array MUST either be a non-empty string or an `Expression` object.
```
(assuming adherence to [ECMA-402 precedent](https://tc39.es/ecma402/#sec-partitionpattern))